src/timepicker/__tests__/time-picker.scenario.js

```diff
@@ -42,8 +50,8 @@ const Controlled = ({
         size={size}
         {...restProps}
       />
-      <p data-e2e="hours">hour: {time ? time.getHours() : 'null'}</p>
-      <p data-e2e="minutes">minute: {time ? time.getMinutes() : 'null'}</p>
+      <p data-e2e="hours">hour: {time ? getHours(time) : 'null'}</p>
```

Comment:
Will methods on the `Date` class continue to work as normal if no adapter is used? Just confirming, but was this change  made to avoid code duplication? It may be worth separating to assert existing code patterns continue to function as-expected

Reply:
This change was made so that this function would work correctly for moment version in the test. I could split it out, but it was kind of nice to be able to just re-run the original test fully. 

And yah, it behaves the exact same way for the standard version.  It's making the same calls inside of `utils`

Reply:
So, I spent a little bit more time digging into this null type issue, and I'm a little stuck on it. 

`const seconds = dateToSeconds(utils.date(), utils);`

That call is a prime example of the problem I'm hitting. 

If the argument passed into `date` is null, it will just return null, otherwise it will return a date.  This means the return type has to be `Date | null` Handling for this gave me two options, but of which seemed bad. 

**1. Always refine `utils.date`**

```
const seconds = dateToSeconds(utils.date() || new Date(), utils);
```

This solves the problem of refining the date down to the right type I needed for `dateToSeconds` but to do it I have to introduce this super weird condition that can't ever be reached.  The second part of that expression only exists because flow's type system isn't currently smart enough to know that the returned object will always be there. 

**2. Always handle null in date processing functions**

I could change it to be able to handle that null value, and would fix the type system, but this also seems not great.  I know that `dateToSeconds` is expecting an explicit date(because date() is being passed), but the only way to make it work is to handle a null value that can never happen. 

I also explored somehow wrapping `date` with `%checks` functions that the type system could figure out that `date` was returning a date object if you passed it undefined, but all of those solutions still require some kind of if statement for the refining to happen, and so we're back in the same situation as the first bad option above. 

I think your concern about null types not being handled correctly is totally valid, I just think that this solution as it is might cover it.  

Passing null to `date` will throw a flow error in a case when it's actually okay to pass null.  The problem here is more that we're preventing people from doing something that would potentially be convenient, rather than allowing them to do something potentially harmful.  Given that, I think this is maybe an acceptable tradeoff. 

If you disagree, the `dateToSeconds(utils.date() || new Date(), utils)` style call everywhere we're calling `date` seems like next best to me. 

Thanks for the super prompt and clear feedback on this PR

Reply:
@chasestarr Going back and reading my comment and this comment it was referring to 
> My worry is that we will miss handling the null case and cause a fairly common pattern, initializing controlled state variables to null, to break amongst different date-io adapters

I assumed you were talking about my decision to just act like the ability to pass null and get back null in date just didn't exist, but I'm not sure that's the case. The two links you include in your comment don't seem to have linked to the lines in question correctly, they just go to the whole file.  Could you update those so I can follow better what you mean?